ROADMAP.md

```diff
@@ -69,18 +69,73 @@ There's a loose collection of some BigchainDB 2.0 "TODO" items in the GitHub org
 
 ## Some Goals for BigchainDB 2.1 and Beyond
 
-- Document how to run a production validator so that it's protected from denial-of-service (DoS) attacks and similar issues. See what Tendermint recommends (i.e. "sentry nodes", see issue https://github.com/bigchaindb/bigchaindb/issues/2333).
-- Performance optimization by removing unneeded deepcopy operations.
-- Example projects showing how to use BigchainDB with other decentralized systems, especially Ethereum.
-- Refactoring, especially of the transaction-related code.
+### Short-Term Goals - Low-Hanging Fruits
+
+These are features which are solving a very specific problem. These features, while not scoped, are expected to be small to medium in implementation complexity.
+
+#### Priority 1
+
+- Better node deployment and setup process. Analyze and ask users their issues. Define an action plan (BEP).
+- Privacy using encryption, using plugins
+  - node level encryption
+  - peer to peer encryption on same node - needs KMS
+
+#### Priority 2
+
+- Plugin architecture for transaction validation. Refactor the tx code: BEP-9. Generalize the codebase. Dynamic loading of plugins using validator election.
+- Scale-up and scale-out strategy. Define limits - run infra load tests.
+- Propose a new transaction model that:
+  - doesn't use crypto-conditions
+  - supports the signature scheme used by Ethereum, i.e. secp256k1 keys
+
+#### Priority 3
+
+- Key management support
+
+#### Priority 4
+
+- Ability to delete past transactions. This is an approach to GDPR compliance (e.g. the "right to be forgotten"). If a user has access to keys, let them instigate deletion, otherwise the nodes must vote on deletion. The modified blockchain must still be valid, somehow (when replayed).
+- Scale-up and scale-out strategy. Documentation to educate users why there is a limit. Add another local mempool for capacity (i.e. to moderate the load on the existing mempool).
+- Privacy using zero-knowledge proofs.
+
+### Medium-Term Goals - Journey Projects
+
+These are typically a collection of features which are solving a much broader set of problems. These features can result in several [BigchainDB Enhancement Proposals (BEPs)](https://github.com/bigchaindb/BEPs) during scoping and are expected to be medium to large in implementation complexity.
+
+#### Priority 1
+
+- Support smart contracts (and create new marketing channels) via:
+  - integration with Ethereum
+  - integration with Hyperledger Fabric
+
+#### Priority 2
+
+- Support for light clients.
+
+#### Priority 3
+
+- Asset sharding: give more schema control to users.
+
+### Priority 4
+
+- Asset sharding: Different MongoDB collections for different asset types.
```

Comment:
This is the same as the one above. It can be either in priority 3 or in priority 4.

Reply:
Okay, I deleted it from here and added the word "MongoDB" before "collections" in the version above (under Priority 3).